pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ObjectFunctions.java

```diff
@@ -92,13 +94,40 @@ private static Object coalesceVar(Object... objects) {
     return null;
   }
 
-  @Nullable
-  private static Object coalesce(Object... objects) {
-    for (Object o : objects) {
-      if (o != null) {
-        return o;
+  @ScalarFunction
+  public static Object caseWhen(boolean c1, Object o1, Object oe) {
```

Comment:
operator name is `case`, the relNode function name is `caseWhen`

Reply:
Interesting.. But does that mean these functions won't be recognized by v1 engine because the function name is `case`?

Reply:
not really. the info passed over to the stages contains SqlOperator and function. it will use SqlOperator to lookup first before using function name. thus v1 engine will correctly composed out the transform version of the "case"; and intermediate stage will use the scalar function correctly.